pkg/maker/rawyamltmpl/raw_text_maker.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package rawtexttmpl is a special-case component maker for constructing
+// objects with the assumption that objects have been inlined via RawText into
+// ConfigMaps.
+package rawtexttmpl
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	bundle "github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/apis/bundle/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/converter"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/filter"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/maker"
+)
+
+// Maker makes components via go-templating, for objects stored as raw strings
+// in config maps.
+type Maker struct{}
+
+func (m *Maker) MakeComponent(comp *bundle.ComponentPackage, pm maker.ParamMaker, of *filter.Options) (*bundle.ComponentPackage, error) {
+	comp = comp.DeepCopy()
+	ref := comp.ComponentReference()
+
+	if len(comp.Spec.Objects) == 0 {
+		return nil, fmt.Errorf("no objects found for component %v", ref)
+	}
+
+	// Filter to only get the raw text config maps.
+	objs := filter.Filter().Objects(comp.Spec.Objects, &filter.Options{
```

Comment:
What is the desired behavior if `objs` ends up being empty because `comp.Spec.Objects` contains only non-raw-string-config-map objects?

Reply:
yeah, need to fix that https://github.com/GoogleCloudPlatform/k8s-cluster-bundle/issues/102